src/lib.rs

```diff
@@ -1,8 +1,17 @@
 #![cfg_attr(target_os = "none", no_std)]
+#![cfg_attr(target_os = "none", feature(alloc_error_handler))]
+
+extern crate alloc;
+
+// Force `mycelium_alloc` to be linked in, as it provides the allocator.
+#[cfg(target_os = "none")]
+extern crate mycelium_alloc;
```

Comment:
Ah, interesting...I took the approach of just defining a bunch of allocator implementations in the alloc crate, and defining the global allocator in the kernel itself, so we can (eventually) have more policy in the kernel (like a tiered scheme where we switch the allocator once paging is set up).

Reply:
Ah, that's also super reasonable. I just threw it into the `alloc` crate 'cause it was easy to do :p